File: #1957.delete-characters-to-make-fancy-string.cpp
class Solution {
public:
    string makeFancyString(string s) {
        
        string str;
        for (int i = 0; i < s.size(); i++) {
            if (!(s[i] == s[i + 1] && s[i] == s[i + 2])) {
                str += s[i];
            }
        }
        return str;
    }
};

File: #1957.delete-characters-to-make-fancy-string.ts
function makeFancyString(s: string): string {
    let stack: string[] = [];
    for (let i: number = 0; i < s.length; i++) {
        if (stack.length >= 2
            && stack[stack.length - 2] === s[i]
            && stack[stack.length - 1] === s[i]) {
            continue; 
        }
        stack.push(s[i]);
    }
    return stack.join('');
};
